setup.py

```diff
@@ -47,7 +47,7 @@ def version():
         'test': tests_require,
     },
     install_requires=(
-        'azure-identity>=1.7.1,<1.8.0',
+        'azure-identity>=1.16.1',
```

Comment:
/cc @viranch who pinned this dep and the others below it in https://github.com/octodns/octodns-azure/commit/ddca49f975eff7806d7deb70898b0862d509b560

Reply:
Verified the pinned set of requirements works locally so going to unblock by moving forward with this one and then aim to sort out the other clamping in the future. 

```console
(env) coho:octodns-azure ross$ pip freeze | grep azure
azure-common==1.1.28
azure-core==1.30.2
azure-identity==1.17.1
azure-mgmt-core==1.4.0
azure-mgmt-dns==8.1.0
azure-mgmt-privatedns==1.0.0
azure-mgmt-trafficmanager==1.1.0
msrestazure==0.6.4.post1
``

Reply:
Clamping originally came about in https://github.com/octodns/octodns-azure/issues/23#issuecomment-1033444071 due to problems with traffic manager's lib. That was in 2022 though and this seems like a long time to have things clamped for. 🙃 

Realized what I tested with locally didn't include TrafficManager bits so probably didn't hit whatever the clamping was added for. Going to try and test that now.

Reply:
OK. `dynamic` record that hit TM works, but holy 🐄  it takes forever to create even a single item TM record. Updates seem quicker.  Moving forward (again)

Reply:
Came here to realize this is already verified working.